src/machines/lesson-player-machine.ts

```diff
@@ -27,83 +32,147 @@ export type PlayerStateEvent =
-        PAUSE: 'paused',
+      loaded: {
+        entry: [
+          assign({
+            lesson: (_, event: any) => event.lesson,
+          }),
+        ],
+        on: {
+          PLAY: 'playing',
+          VIEW: 'viewing',
+          SUBSCRIBE: 'subscribing',
+          JOIN: 'joining',
+          PAUSE: 'paused',
+        },
       },
-    },
-    playing: {
-      on: {
-        QUIZ: 'quizzing',
-        COMPLETE: 'completed',
-        RATE: 'rating',
-        PAUSE: 'paused',
+      viewing: {
+        entry: ['sendTelemetry'],
+        on: {
+          PLAY: 'playing',
+        },
       },
-    },
-    paused: {
-      on: {
-        PLAY: 'playing',
+      playing: {
+        entry: ['sendTelemetry'],
+        on: {
+          QUIZ: 'quizzing',
+          COMPLETE: 'completed',
+          RATE: 'rating',
+          PAUSE: 'paused',
+        },
       },
-    },
-    completed: {
-      on: {
-        NEXT: 'showingNext',
-        SUBSCRIBE: 'subscribing',
-        JOIN: 'joining',
-        RATE: 'rating',
-        QUIZ: 'quizzing',
+      paused: {
+        entry: ['sendTelemetry'],
+        on: {
+          PLAY: 'playing',
+        },
       },
-    },
-    quizzing: {
-      on: {
-        NEXT: 'showingNext',
-        SUBSCRIBE: 'subscribing',
-        JOIN: 'joining',
-        RATE: 'rating',
+      completed: {
+        entry: ['sendTelemetry'],
+        on: {
+          NEXT: 'showingNext',
+          SUBSCRIBE: 'subscribing',
+          JOIN: 'joining',
+          RATE: 'rating',
+          QUIZ: 'quizzing',
+        },
       },
-    },
-    rating: {
-      on: {
-        NEXT: 'showingNext',
-        SUBSCRIBE: 'subscribing',
-        JOIN: 'joining',
-        QUIZ: 'quizzing',
+      quizzing: {
+        on: {
+          NEXT: 'showingNext',
+          SUBSCRIBE: 'subscribing',
+          JOIN: 'joining',
+          RATE: 'rating',
+        },
       },
-    },
-    subscribing: {
-      on: {
-        NEXT: 'showingNext',
-        JOIN: 'joining',
-        RATE: 'rating',
-        QUIZ: 'quizzing',
+      rating: {
+        on: {
+          NEXT: 'showingNext',
+          SUBSCRIBE: 'subscribing',
+          JOIN: 'joining',
+          QUIZ: 'quizzing',
+        },
       },
-    },
-    joining: {
-      on: {
-        NEXT: 'showingNext',
-        SUBSCRIBE: 'subscribing',
-        RATE: 'rating',
-        QUIZ: 'quizzing',
+      subscribing: {
+        on: {
+          NEXT: 'showingNext',
+          JOIN: 'joining',
+          RATE: 'rating',
+          QUIZ: 'quizzing',
+        },
+      },
+      joining: {
+        on: {
+          NEXT: 'showingNext',
+          SUBSCRIBE: 'subscribing',
+          RATE: 'rating',
+          QUIZ: 'quizzing',
+        },
+      },
+      showingNext: {
+        on: {
+          LOAD: 'loading',
+        },
       },
     },
-    showingNext: {
-      on: {
-        LOAD: 'loading',
+  },
+  {
+    actions: {
+      sendTelemetry: (context, event) => {
```

Comment:
for specific events we want to record an aXPI verb

![image](https://user-images.githubusercontent.com/86834/95641182-ba5bac00-0a55-11eb-9fc2-8debc83a44d0.png)